longest_palindromic_substring/LongestPalindromicSubstring.java

```diff
@@ -0,0 +1,45 @@
+public class LongestPalindromicSubstring {
+
+    /*
+    This method is used to determine palindrome about a point in string.
+    @param test: Input string whose substrings are to be checked
+    @param left: Left end of result palindromic substring
+    @param right: Right end of result palindromic substring
+    @return: Length of palindromic substring
+    */
+    private static int expandAroundCenter(String test, int left, int right) {
+        int n = test.length();
+        while (left >= 0 && right < n && test.charAt(left) == test.charAt(right)) {
+            left -= 1;
+            right += 1;
+        }
+        return right - left - 1;
+    }
+
+    /*
+    Method to find longest substring which is a palindrome.
+    @param test: Input string whose substrings are to be checked
+    @return: Longest substring of input string which is a palindrome
+
```

Comment:
These blanks are to be kept as in python we keep 2 blank lines. Other implementations also have them.